File: CS/SchedulingDemo.Wpf/Common/SchedulingDemoModule.cs
using System;
using System.Windows;
using DevExpress.DemoData.Models;
using DevExpress.Xpf.Core.Native;
using DevExpress.Xpf.DemoBase;
using DevExpress.Xpf.Scheduling;

namespace SchedulingDemo {
    public class SchedulingDemoModule : DemoModule {
        static SchedulingDemoModule() {
            NWindContext.Preload();
        }

        static ResourceDictionary commonResources;
        static ResourceDictionary CommonResources  { get { return commonResources ?? (commonResources = new ResourceDictionary() { Source = new Uri("/SchedulingDemo;component/Themes/Common.xaml", UriKind.RelativeOrAbsolute) }); } }
        static ResourceDictionary outlookInspiredDemoResources;
        static ResourceDictionary OutlookInspiredDemoResources { get { return outlookInspiredDemoResources ?? (outlookInspiredDemoResources = new ResourceDictionary() { Source = new Uri("/SchedulingDemo;component/Themes/OutlookInspired.xaml", UriKind.RelativeOrAbsolute) }); } }

        bool addCommonResources;
        bool addOutlookInspiredDemoResources;
        public SchedulingDemoModule(bool addCommonResources = true, bool addOutlookInspiredDemoResources =false) {
            this.addCommonResources = addCommonResources;
            this.addOutlookInspiredDemoResources = addOutlookInspiredDemoResources;
            if (addCommonResources)
                Resources.MergedDictionaries.Add(CommonResources);
            if(addOutlookInspiredDemoResources)
                Resources.MergedDictionaries.Add(OutlookInspiredDemoResources);
            Loaded += OnLoaded;
        }

        protected virtual bool NeedChangeEditorsTheme { get { return false; } }
        SchedulerControl Scheduler { get; set; }
        
        protected virtual void LoadDemoData() {
        }

        protected override void Clear() {
            base.Clear();
            Scheduler.DataSource = null;
        }

        protected override void OnInitialized(EventArgs e) {
            base.OnInitialized(e);
            if (this.addCommonResources && !Resources.MergedDictionaries.Contains(CommonResources))
                throw new InvalidOperationException();
            if(this.addOutlookInspiredDemoResources && !Resources.MergedDictionaries.Contains(OutlookInspiredDemoResources))
                throw new InvalidOperationException();
        }

        void OnLoaded(object sender, RoutedEventArgs e) {            
            Scheduler = LayoutHelper.FindElementByType<SchedulerControl>(this);
        }
    }
}

File: VB/SchedulingDemo.Wpf/Common/SchedulingDemoModule.vb
Imports System
Imports System.Windows
Imports DevExpress.DemoData.Models
Imports DevExpress.Xpf.Core.Native
Imports DevExpress.Xpf.DemoBase
Imports DevExpress.Xpf.Scheduling

Namespace SchedulingDemo
    Public Class SchedulingDemoModule
        Inherits DemoModule

        Shared Sub New()
            NWindContext.Preload()
        End Sub


        Private Shared commonResources_Renamed As ResourceDictionary
        Private Shared ReadOnly Property CommonResources() As ResourceDictionary
            Get
                If commonResources_Renamed IsNot Nothing Then
                    Return commonResources_Renamed
                Else
                    commonResources_Renamed = New ResourceDictionary() With {.Source = New Uri("/SchedulingDemo;component/Themes/Common.xaml", UriKind.RelativeOrAbsolute)}
                    Return commonResources_Renamed
                End If
            End Get
        End Property

        Private Shared outlookInspiredDemoResources_Renamed As ResourceDictionary
        Private Shared ReadOnly Property OutlookInspiredDemoResources() As ResourceDictionary
            Get
                If outlookInspiredDemoResources_Renamed IsNot Nothing Then
                    Return outlookInspiredDemoResources_Renamed
                Else
                    outlookInspiredDemoResources_Renamed = New ResourceDictionary() With {.Source = New Uri("/SchedulingDemo;component/Themes/OutlookInspired.xaml", UriKind.RelativeOrAbsolute)}
                    Return outlookInspiredDemoResources_Renamed
                End If
            End Get
        End Property

        Private addCommonResources As Boolean
        Private addOutlookInspiredDemoResources As Boolean
        Public Sub New(Optional ByVal addCommonResources As Boolean = True, Optional ByVal addOutlookInspiredDemoResources As Boolean =False)
            Me.addCommonResources = addCommonResources
            Me.addOutlookInspiredDemoResources = addOutlookInspiredDemoResources
            If addCommonResources Then
                Resources.MergedDictionaries.Add(CommonResources)
            End If
            If addOutlookInspiredDemoResources Then
                Resources.MergedDictionaries.Add(OutlookInspiredDemoResources)
            End If
            AddHandler Loaded, AddressOf OnLoaded
        End Sub

        Protected Overridable ReadOnly Property NeedChangeEditorsTheme() As Boolean
            Get
                Return False
            End Get
        End Property
        Private Property Scheduler() As SchedulerControl

        Protected Overridable Sub LoadDemoData()
        End Sub

        Protected Overrides Sub Clear()
            MyBase.Clear()
            Scheduler.DataSource = Nothing
        End Sub

        Protected Overrides Sub OnInitialized(ByVal e As EventArgs)
            MyBase.OnInitialized(e)
            If Me.addCommonResources AndAlso (Not Resources.MergedDictionaries.Contains(CommonResources)) Then
                Throw New InvalidOperationException()
            End If
            If Me.addOutlookInspiredDemoResources AndAlso (Not Resources.MergedDictionaries.Contains(OutlookInspiredDemoResources)) Then
                Throw New InvalidOperationException()
            End If
        End Sub

        Private Sub OnLoaded(ByVal sender As Object, ByVal e As RoutedEventArgs)
            Scheduler = LayoutHelper.FindElementByType(Of SchedulerControl)(Me)
        End Sub
    End Class
End Namespace
